### PR TITLE
Add configurable max initialization attempts

### PR DESCRIPTION
**Pull request by micah-jaffe**
Match KCL 2.x configurable max initialization attempts. This will help speed up long-running unit tests introduced by shard sync optimizations.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java [hunk 1]
```diff
@@ -435,6 +435,8 @@ public final void testInitializationFailureWithRetries() throws Exception {
         String stageName = "testInitializationWorker";
         IRecordProcessorFactory recordProcessorFactory = new TestStreamletFactory(null, null);
         config = new KinesisClientLibConfiguration(stageName, null, null, WORKER_ID);
+        config.withMaxInitializationAttempts(2);
```
Comment (ychunxue) [inline src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java:439]: In setup(), we initialize the global config, which is used by most of the test cases. Shall we also set the max attempts for this global config?

**Review by ychunxue — COMMENTED** [review]
Minor comment

File: src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java (hunk 1 above)
Comment (micah-jaffe) [reply src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/WorkerTest.java:439]: Sure, I'll set a global config to 1 and override it where appropriate (only this unit test currently)

**Review by ychunxue — APPROVED** [review]
LGTM

**Pull request merged**